java-frontend/src/main/java/org/sonar/plugins/java/api/DependencyVersionAware.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonar.plugins.java.api;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public interface DependencyVersionAware {
+
+  boolean isCompatibleWithDependencies(Function<String, Optional<Version>> dependencyFinder);
```

Comment:
Since the usage of this method is not super straight-forward I would document it with a couple of example on how to implement it.